apis/gasbuddy.js

```diff
@@ -0,0 +1,255 @@
+    });
+
+    return stations;
+}
+
+/**
+ * @function getAllStations
+ * @description Requests all station and fuel types as paginated requests.
+ * @async
+ *
+ * @returns {Object[]} Returns object described in the provider documentation.
+ */
+async function getAllStations() {
+    const promises = config.types.reduce((acc, type) => {
+        const paths = getRequestPaths(type);
+        paths.forEach(path => acc.push(fetchPaginatedStations(type, path)));
+        return acc;
+    }, []);
+
+    const responses = await Promise.all(promises);
+
+    return responses.flat();
+}
+
+/**
+ * @function mergePrices
+ * @description Merges fuel prices of different types of gas station
+ *
+ * @param {Object[]} responses - List of gas stations with prices of single fuel type.
+ *
+ * @returns {Object} Returns gas stations with merged prices and max prices per fuel type.
+ */
+function mergePrices(responses) {
+    const { indexedStations, maxPricesByType } = responses.reduce(({ indexedStations, maxPricesByType }, station) => {
+        const stationKey = `${station.name}-${station.address}`;
+
+        if (!indexedStations[stationKey]) {
+            indexedStations[stationKey] = station;
+        } else {
+            indexedStations[stationKey].prices[station.fuelType] = station.prices[station.fuelType];
+        }
+
+        if (!maxPricesByType[station.fuelType] || maxPricesByType[station.fuelType] < station.prices[station.fuelType]) {
+            maxPricesByType[station.fuelType] = station.prices[station.fuelType];
+        }
+
+        return { indexedStations, maxPricesByType };
+    }, { indexedStations: {}, maxPricesByType: {} });
+
+    return { stations: Object.values(indexedStations), maxPricesByType };
+}
+
+/**
+ * @function getData
+ * @description Performs the data query and processing.
+ * @async
+ *
+ * @returns {Object} Returns object described in the provider documentation.
+ *
+ * @see apis/README.md
+ */
+async function getData() {
+    const responses = await getAllStations();
+
+    const { stations, maxPricesByType } = mergePrices(responses);
+
+    stations.forEach(station => fillMissingPrices(station, maxPricesByType));
+
+    const filteredStations = stations.filter(station => station.distance <= config.radius);
+
+    const stationsSortedByDistance = filteredStations.sort((a, b) => a.distance - b.distance);
+    const stationsSortedByPrice = [...stationsSortedByDistance].sort(sortByPrice);
+    return {
+        types: ['regular', 'midgrade', 'premium', 'diesel', 'e85', 'unl88'],
+        unit: 'mile',
+        currency: 'USD',
+        byPrice: stationsSortedByPrice,
+        byDistance: stationsSortedByDistance
+    };
+}
+
+/**
+ * @module apis/gasbuddy
+ * @description Queries data from https://www.gasbuddy.com
+ *
+ * @requires external:node-fetch
+ * @requires external:node-html-parser
+ *
+ * @param {Object} options - Configuration.
+ * TODO: Convert this from radius to lastupdated or something.
```

Comment:
Instead of having a TODO here, it would be better to create an issue in the repository